### PR TITLE
Added a dynamic gameboard creator in Board class and a Board builder …

### PR DESCRIPTION
**Pull request by ericdrosado**
…in UI class based on chosen board dimension.

### CODE REVIEW

File: TicTacToe/TicTacToe/ComputerLogic.cs [hunk 1]
```diff
@@ -29,59 +51,49 @@ public class ComputerLogic {
...
+            int score;
             if (marker == "O" && this.winConditions.IsWinner(gameBoard)) {
-                move.Score = 1000;
+                score = 1000 - depth;
             } else if (marker == "X" && this.winConditions.IsWinner(gameBoard)) {
-                move.Score = -1000;
+                score = depth - 1000;
             } else {
-                move.Score = 0;
+                score = 0;
```
Comment (cdemyanovich) [inline TicTacToe/TicTacToe/ComputerLogic.cs]: `1000` and `0` could use some good names to give them meaning.

File: TicTacToe/TicTacToe/ComputerLogic.cs [hunk 2]
```diff
@@ -29,59 +51,49 @@ public class ComputerLogic {
...
-                for (int i = 0; i < moves.Count; i++) {
-                    if (moves[i].Score > bestMoveScore) {
-                        bestMoveScore = moves[i].Score;
-                        bestMove = i;
-                    }
+                for (int i = 0; i < availableSpaces.Count; i++) {
+                    string[] gameBoardCopy = (string[]) gameBoard.Clone();
+                    gameBoardCopy[availableSpaces.ElementAt(i)] = marker;
+                    int score = GetScore(gameBoardCopy, alpha, beta, AlternateMarker(marker), depth++);
+                    alpha = Math.Max(alpha, score );
```
Comment (cdemyanovich) [inline TicTacToe/TicTacToe/ComputerLogic.cs]: Extra space between `score` and `)`.

File: TicTacToe/TicTacToe/ComputerLogic.cs [hunk 3]
```diff
@@ -29,59 +51,49 @@ public class ComputerLogic {
...
+                    if (beta <= alpha || depth == 15) {
```
Comment (cdemyanovich) [inline TicTacToe/TicTacToe/ComputerLogic.cs]: What does `15` mean?

File: TicTacToe/TicTacToe/ComputerLogic.cs [hunk 4]
```diff
@@ -29,59 +51,49 @@ public class ComputerLogic {
...
+                for (int i = 0; i < availableSpaces.Count; i++) {
+                    string[] gameBoardCopy = (string[]) gameBoard.Clone();
+                    gameBoardCopy[availableSpaces.ElementAt(i)] = marker;
+                    int score = GetScore(gameBoardCopy, alpha, beta, AlternateMarker(marker), depth++);
+                    beta = Math.Min(beta, score);
+                    if (beta <= alpha || depth == 15) {
+                        break;
+                    }  
                 }
+                return beta;
```
Comment (cdemyanovich) [inline TicTacToe/TicTacToe/ComputerLogic.cs]: How could you eliminate the duplication between the bodies of the `if` and `else`?

Also, there's that `15` again.

File: TicTacToe/TicTacToe/UI.cs [hunk 5]
```diff
@@ -43,20 +58,35 @@ public class UI {
...
-            "                       " + gameBoard[3] + " | " + gameBoard[4] + " | " + gameBoard[5] +
-            "\n                      " + "---+---+---\n" +
-            "                       " + gameBoard[6] + " | " + gameBoard[7] + " | " + gameBoard[8] + "\n";
+        public string BuildBoard(string[] gameBoard, int boardSize) {
+            string board = "";
+            for (int index = 0; index <= boardSize * boardSize - boardSize; index += boardSize) {
+                string row = BuildRow(gameBoard, boardSize, index);
+                board = string.Concat(board, row);
+                if (index < boardSize * boardSize - boardSize) {
+                    board = boardSize == 4? string.Concat(board, "\n                      " + "---+---+---+---\n"):
```
Comment (cdemyanovich) [inline TicTacToe/TicTacToe/UI.cs]: Why the check for `4` here?

File: TicTacToe/TicTacToe/UI.cs [hunk 6]
```diff
@@ -43,20 +58,35 @@ public class UI {
...
+                if (index < boardSize * boardSize - boardSize) {
+                    board = boardSize == 4? string.Concat(board, "\n                      " + "---+---+---+---\n"): 
+                        string.Concat(board, "\n                      " + "---+---+---\n");
+                }
+            }
+            return board;
+        }
+
+        private string BuildRow(string[] gameBoard, int boardSize, int index) {
+            string row = index > 9? "                      " + gameBoard[index]:
```
Comment (cdemyanovich) [inline TicTacToe/TicTacToe/UI.cs]: Why the check for `9` here?

File: TicTacToe/TicTacToe/WinConditions.cs [hunk 7]
```diff
@@ -41,7 +41,7 @@ public class WinConditions {
             for (int i = 0; i < boardDimension; i++) {
```
Comment (tommcgee) [inline TicTacToe/TicTacToe/WinConditions.cs]: Would be nice to see some more meaningful names than `i` and `j`

**Pull request merged**